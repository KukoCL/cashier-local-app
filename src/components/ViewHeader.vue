<template>
  <div class="view-header">
    <h1>{{ title }}</h1>
  </div>
</template>

<script setup lang="ts">
defineProps<{
  title: string;
}>();
</script>

<style scoped>
.view-header {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 2rem;
}

.view-header h1 {
  margin: 0;
  color: #ffffff;
}
</style>
