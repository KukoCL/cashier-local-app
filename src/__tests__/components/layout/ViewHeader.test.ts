import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ViewHeader from '../../../components/ViewHeader.vue'

describe('ViewHeader', () => {
  it('should render with required title prop', () => {
    const title = 'Test Page Title'
    const wrapper = mount(ViewHeader, {
      props: {
        title,
      },
    })

    expect(wrapper.find('h1').text()).toBe(title)
    expect(wrapper.classes()).toContain('view-header')
  })

  it('should render different titles correctly', () => {
    const testCases = [
      'Home Page',
      'Products List',
      'Create Product',
      'User Settings',
      'Dashboard Overview',
    ]

    testCases.forEach((title) => {
      const wrapper = mount(ViewHeader, {
        props: {
          title,
        },
      })

      expect(wrapper.find('h1').text()).toBe(title)
    })
  })

  it('should render empty title when provided', () => {
    const wrapper = mount(ViewHeader, {
      props: {
        title: '',
      },
    })

    expect(wrapper.find('h1').text()).toBe('')
    expect(wrapper.find('h1').exists()).toBe(true)
  })

  it('should render title with special characters', () => {
    const specialTitle = 'Título con Ñ & Símbolos - 123!'
    const wrapper = mount(ViewHeader, {
      props: {
        title: specialTitle,
      },
    })

    expect(wrapper.find('h1').text()).toBe(specialTitle)
  })

  it('should render title with HTML entities correctly', () => {
    const titleWithEntities = 'Products & Services'
    const wrapper = mount(ViewHeader, {
      props: {
        title: titleWithEntities,
      },
    })

    expect(wrapper.find('h1').text()).toBe(titleWithEntities)
  })

  it('should handle long titles', () => {
    const longTitle = 'This is a very long title that might need to be handled differently in the UI ' +
      'but should still render correctly in the component'
    const wrapper = mount(ViewHeader, {
      props: {
        title: longTitle,
      },
    })

    expect(wrapper.find('h1').text()).toBe(longTitle)
  })

  it('should maintain accessibility with proper heading structure', () => {
    const wrapper = mount(ViewHeader, {
      props: {
        title: 'Accessible Title',
      },
    })

    const h1Element = wrapper.find('h1')
    expect(h1Element.exists()).toBe(true)
    expect(h1Element.text()).toBe('Accessible Title')
  })
})
