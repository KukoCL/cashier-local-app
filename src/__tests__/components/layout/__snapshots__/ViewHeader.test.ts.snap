// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewHeader > should have correct HTML structure 1`] = `
"<div data-v-66952daf="" class="view-header">
  <h1 data-v-66952daf="">Test Title</h1>
</div>"
`;
